linera-views/src/queue_view.rs

```diff
@@ -60,31 +60,32 @@ where
             context,
             stored_indices,
             front_delete_count: 0,
-            was_cleared: false,
+            delete_storage_first: false,
             new_back_values: VecDeque::new(),
             stored_hash: hash,
             hash: Mutex::new(hash),
         })
     }
 
     fn rollback(&mut self) {
-        self.was_cleared = false;
+        self.delete_storage_first = false;
         self.front_delete_count = 0;
         self.new_back_values.clear();
         *self.hash.get_mut() = self.stored_hash;
     }
 
     fn flush(&mut self, batch: &mut Batch) -> Result<(), ViewError> {
-        let mut save_stored_indices = false;
+        if self.delete_storage_first {
+            batch.delete_key_prefix(self.context.base_key());
+            self.stored_hash = None;
+        }
```

Comment:
I wonder if `self.stored_indices = Range::default();` would help here in the same way as `self.stored_hash = None` did.

Reply:
I tried and it did not.

Reply:
ok